_posts/2021-05-04-how-to-update-your-ref-arch.adoc

```diff
@@ -0,0 +1,490 @@
+inputs = {
+  # Many modules require these two inputs, so we set them globally here to keep all the child terragrunt.hcl files more
+  # DRY
+  aws_account_id = local.account_id
+  aws_region     = local.aws_region
+}
+....
++
+[NOTE]
+.Explanation
+====
+In the Reference Architecture v1.0, we had a root configuration for each account to ensure
+that we can create a different state bucket for each account. While this isn't necessary for the Service Catalog, we
+switch to a single root `terragrunt.hcl` config here (which is possible due to the advanced functions that are available
+in newer Terragrunt versions) because there are more common blocks that are necessary, and we want to keep these
+blocks DRY.
+
+To support the new requirements of the Service Catalog, we also introduce two code generation configuration:
+
+* `generate "provider"`: Uses the terragrunt
+  https://terragrunt.gruntwork.io/docs/features/keep-your-terraform-code-dry/#dry-common-terraform-code-with-terragrunt-generate-blocks[code
+  generation] feature to inject the provider block into the module prior to invoking terraform.
+* `generate` attribute of the `remote_state` block: Similar to the provider block generation, this attribute injects the
+  `terraform.backend` configuration.
+
+Finally, we introduce a `locals` block to define references that can be reused throughout the configuration. Note that
+for new commonly used variables, we use `read_terragrunt_config` instead of `yamldecode(file())` to allow for use of
+terragrunt functions in the config.
+
+Note that the suggested config continues to use the `region.yaml` data file to simplify the migration process. You can
+optionally update this file to `hcl` for consistency.
+====
++
+. Create new data files for the root config (these are the files that are read in in the `locals` blocks):
+* In the root of the `infrastructure-live` repository, add a `common.hcl` file with the following contents:
++
+....
+locals {
+  # TODO: A unique name prefix to set for all infrastructure resources created in your accounts.
+  name_prefix = ""
+  # TODO: the default AWS region to use. This should be the same as where the terraform state S3 bucket is
+  # currently provisioned.
+  default_region = ""
+}
+....
+
+* In each account folder (e.g., `infrastructure-live/dev` or `infrastructure-live/shared`), add a file named
+  `account.hcl` with the following contents:
++
+....
+locals {
+  # TODO: Update with the actual information for each account
+  # The user friendly name of the AWS account. Usually matches the folder name.
+  account_name = ""
+  # The 12 digit ID number for your AWS account.
+  account_id = ""
+}
+....
+
+* (optional) If you wish to replace your yaml variable files with HCL, in each region folder (e.g.,
+  `infrastructure-live/dev/us-east-2`), add a file named `region.hcl` with the following contents:
++
+....
+locals {
+  # TODO: enter the region to use for all resources in this subfolder.
+  aws_region = ""
+}
+....
```

Comment:
this is a pretty easy step to follow, so just make it non optional?

Reply:
I don't feel too strongly about it, but weighing it, I think we should keep it optional, in case people don't want to do this. Yes, it's easy, but I don't want to give the impression that it's required. ⚖️ I could go either way on it, really.